elastic/eui
# build: Cypress React 18 support and React version switcher

## Summary

Add @cypress/react18 renderer and implement React version switching logic to run Cypress tests against any version of react supported by EUI.

Failing tests when running cypress using React 18 will be addressed in a separate PR.

## QA

Please note that this PR will fail in CI due to wrong types and unit test issues related to React 18

1. Checkout this branch locally
2. Run `yarn` to install new packages
3. Run `yarn test-cypress --react-version 17` (or `--react-version 16`) and see it passing all tests
4. Run `yarn test-cypress` and see some tests failing because of React 18 differences

### General checklist
